### PR TITLE
Fix UX consistency 

### PR DESCRIPTION
**Pull request by AlexEvanczuk**
# Summary (please read before review)

This fixes the issue from https://github.com/BigRails/danger-packwerk/issues/2

TLDR: `danger-packwerk` can only leave GitHub comments where there is a visible diff in GitHub (this is a GitHub restriction). Since we don't want to leave top-level PR comments, this makes it so certain new violation additions can never show up in Github. Namely, if we add a new violation against a constant with existing violations, the existing constant may not be part of the diff.

I have two implementations, as separate commits, to fix this in this PR:
1) I toggle between leaving the comment on the constant OR the file, with a preference for the constant if it's in the diff. I like this becuase I feel like the comment on the constant draws attention to that *constant* being the focus of attention. Although note the message will always contain information about the constant. I don't like this approach because its a lot more complex (leaving room for error and harder maintenance) and also exposes more in the public API (although I could probably address this with a different design).
2) I simply *always* leave the comment on the new file. This simplifies things and is also a more consistent experience for the user. It simplifies the solution a bit too.

### Do you prefer solution (1) or solution (2)? Let me know!

# Commits
- clean up common test setup
- clean up let blocks
- remove unnecessary context
- add test case for single constant addition
- add test for new reference against existing constant
- fix test by changing behavior
- add another test case
- switch to simpler implementation
- Revert "switch to simpler implementation"

### CODE REVIEW

**Review by shageman — APPROVED** [review]
Simplicity of implementation pushes me to solution #2

**Comment by AlexEvanczuk** [comment]
> Simplicity of implementation pushes me to solution #2

Sounds good @shageman . I've dropped the revert of solution 2 and force pushed along with a commit to bump the version.
